### PR TITLE
Functional: Fix Mypy Issue

### PR DESCRIPTION
**Pull request by fthaler**
## Description

Fixes a Mypy issue which appeared due to the previous PR.

## Requirements

Before submitting this PR, please make sure:

- [x] You have run the code checks, tests and documentation build successfully
- [x] All fixes and all new functionality are tested and documentation is up to date
- [x] You looked at the [review checklist](https://github.com/GridTools/gt4py/blob/master/REVIEW_CHECKLIST.txt)

Additionally, if this PR contains code authored by new contributors:

- [x] All the authors are covered by a valid contributor assignment agreement,
signed by the employer if needed, provided to ETH Zurich
- [x] The names of all the new contributors have been added to an updated
version of the AUTHORS.rst file included in the PR

### CODE REVIEW

File: src/functional/ffront/foast_passes/type_deduction.py [hunk 1]
```diff
@@ -348,6 +348,7 @@ def _visit_neighbor_sum(self, node: foast.Call, **kwargs) -> foast.Call:
 
     def _visit_broadcast(self, node: foast.Call, **kwargs) -> foast.Call:
         field_type = cast(ct.FieldType, node.args[0].type)
+        assert isinstance(node.args[1], foast.TupleExpr)
         broadcast_dims_expr = node.args[1].elts
```
Comment (tehrengruber) [inline src/functional/ffront/foast_passes/type_deduction.py:352]:
```suggestion
        broadcast_dims_expr = cast(node.args[1], foast.TupleExpr).elts
```
This is ensured by `is_callable` function and we do not place any asserts that fail anyway if an invariant is violated.

**Review by tehrengruber — CHANGES_REQUESTED** [review]

File: src/functional/ffront/foast_passes/type_deduction.py (hunk 1 above)
Comment (fthaler) [reply src/functional/ffront/foast_passes/type_deduction.py:352]: But the Mypy (and thus the pre-commit hooks and thus CI) fails without this.

Comment (fthaler) [reply src/functional/ffront/foast_passes/type_deduction.py:352]: https://github.com/GridTools/gt4py/runs/6706727049?check_suite_focus=true

Comment (tehrengruber) [reply src/functional/ffront/foast_passes/type_deduction.py:352]: I would be really surprised if this fails after you have added the cast that I suggested. Am I missing something here?

Comment (fthaler) [reply src/functional/ffront/foast_passes/type_deduction.py:352]: Ah sorry I somehow missed the suggestion!

**Review by tehrengruber — APPROVED** [review]

**Pull request merged**